File: android/src/main/kotlin/com/umair/mapbox_navigation/models/MapBoxEventResults.kt
package com.umair.mapbox_navigation.models

data class MapBoxEventResults(var eventName: String, var data: String) {

    override fun toString(): String {
        return "{" +
                "  \"eventName\": \"$eventName\"," +
                "  \"data\": $data" +
                "}"
    }
}

File: lib/mapbox_event_results.dart
import 'dart:convert';

class MapBoxEventResults {
  String eventName;
  String data;

  MapBoxEventResults({this.eventName, this.data});

  MapBoxEventResults.fromJson(Map<String, dynamic> json) {
    eventName = json['eventName'];
    data = jsonEncode(json['data']);
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['eventName'] = this.eventName;
    data['data'] = this.data;
    return data;
  }
}
